.github/workflows/nfr.yml

```diff
@@ -0,0 +1,196 @@
+        tags: ${{ steps.nginx-meta.outputs.tags }}
+        context: "."
+        platforms: linux/amd64
+        provenance: false
+        pull: true
+        push: true
+        build-args: |
+          NJS_DIR=internal/mode/static/nginx/modules/src
+          NGINX_CONF_DIR=internal/mode/static/nginx/conf
+          BUILD_AGENT=gha
+        secrets: |
+          ${{ inputs.nginx-plus == true && format('"nginx-repo.crt={0}"', secrets.NGINX_CRT) || '' }}
+          ${{ inputs.nginx-plus == true && format('"nginx-repo.key={0}"', secrets.NGINX_KEY) || '' }}
+
+    - name: 'Setup dotenv file'
+      working-directory: ./tests/scripts
+      env:
+        GKE_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+        SERVICE_ACCOUNT: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+        NODES_SERVICE_ACCOUNT: ${{ secrets.GKE_NODES_SERVICE_ACCOUNT }}
+      run: |
+        echo "RESOURCE_NAME=nfr-tests-${{ github.run_id }}" >> vars.env
+        echo "TAG=${{ inputs.version }}-${{ github.run_id }}" >> vars.env
+        echo "PREFIX=gcr.io/${GKE_PROJECT_ID}/ngf-nfr/nginx-gateway-fabric" >> vars.env
+        echo "NGINX_PREFIX=gcr.io/${GKE_PROJECT_ID}/ngf-nfr/nginx-gateway-fabric/nginx" >> vars.env
+        echo "NGINX_PLUS_PREFIX=gcr.io/${GKE_PROJECT_ID}/ngf-nfr/nginx-gateway-fabric/nginxplus" >> vars.env
+        echo "GKE_CLUSTER_NAME=nfr-tests-${{ github.run_id }}" >> vars.env
+        echo "GKE_CLUSTER_ZONE=us-east1-b" >> vars.env
+        echo "GKE_CLUSTER_REGION=us-east1" >> vars.env
+        echo "GKE_PROJECT=${GKE_PROJECT_ID}" >> vars.env
+        echo "GKE_SVC_ACCOUNT=${SERVICE_ACCOUNT}" >> vars.env
+        echo "GKE_NODES_SERVICE_ACCOUNT=${NODES_SERVICE_ACCOUNT}" >> vars.env
+        echo "IMAGE=projects/debian-cloud/global/images/debian-11-bullseye-v20240213" >> vars.env
+        echo "NETWORK_TAGS=nfr-tests-${{ github.run_id }}" >> vars.env
+        echo "NGF_REPO=nginxinc" >> vars.env
+        echo "NGF_BRANCH=${{ github.ref_name }}" >> vars.env
+        echo "SOURCE_IP_RANGE=$(curl -sS -4 icanhazip.com)/32" >> vars.env
+        echo "ADD_VM_IP_AUTH_NETWORKS=true" >> vars.env
+        echo "PLUS_ENABLED=${{ inputs.nginx-plus }}" >> vars.env
+        echo "GINKGO_LABEL=" >> vars.env
+
+    - name: Create GKE cluster
+      working-directory: ./tests
+      run:
+        make create-gke-cluster
+
+    - name: Setup VM and router
+      working-directory: ./tests
+      run:
+        make create-and-setup-vm
+
+    - name: Run Performance Tests
+      working-directory: ./tests
+      if: ${{ inputs.test_label == 'performance' || inputs.test_label == 'all' }}
+      run:
+        sed -i '/^GINKGO_LABEL=/s/=.*/="performance"/' "scripts/vars.env" && make run-tests-on-vm
+
+    - name: Run Upgrade Tests
+      working-directory: ./tests
+      if: ${{ inputs.test_label == 'upgrade' || inputs.test_label == 'all' }}
+      run:
+        sed -i '/^GINKGO_LABEL=/s/=.*/="upgrade"/' "scripts/vars.env" && make run-tests-on-vm
```

Comment:
Can we just use the Github var directly in the command instead of creating multiple jobs? This won't scale as we add more tests.

Reply:
Yeah that makes sense. I'll change it.